test/07-validation.spec.ts

```diff
@@ -61,19 +61,24 @@ describe('validation', () => {
 			expect(validation.checkInt('200')).to.equal(200);
 			expect(validation.checkInt('0')).to.equal(0);
 			expect(validation.checkInt('-3')).to.equal(-3);
+			expect(validation.checkInt(50.55)).to.equal(50); // Fractional digits
+			expect(validation.checkInt('0x11')).to.equal(17); // Hexadecimal
```

Comment:
I don't think we should add this extra input support. The supervisor expects integers to be provided in decimal, and I don't want any unexpected behaviour by supporting anything else.

Reply:
Do we want to return undefined for the above cases mentioned (fractional, hex, oct, binary) ? Or leave the current code and remove these test cases as they aren't a specification we are trying to support.

Reply:
Covering them is still a good idea I think, so we should return undefined, and let that be handled downstream

Reply:
Since you mentioned to let values that aren't Integers or can't be parsed into one without data lose be handled downstream I have added these cases to the test suite that makes sure they return undefined.